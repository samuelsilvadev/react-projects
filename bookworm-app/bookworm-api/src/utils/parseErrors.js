module.exports = function parseErrors(errors) {
	const result = {};
	if (errors) {
		Object
			.entries(errors)
			.forEach((value) => {
				result[value[0]] = value[1].message;
			});
	}
	return result;
};
