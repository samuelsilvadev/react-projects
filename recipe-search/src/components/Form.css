.search-form {
	display: flex;
	justify-content: center;
	margin: 10px 0;
}

.search-form__input,
.search-form__button {
	font-size: 16px;
	font-weight: 400;
	padding: 6px 12px;
}

.search-form__input {
	background-color: #fff;
	border: 1px solid #ced4da;
	border-radius: 4px;
	color: #495057;
	height: 35px;
	line-height: 1.5;
	margin-right: 10px;
}

.search-form__button {
	background-color: #ced4da;
	border: none;
}
