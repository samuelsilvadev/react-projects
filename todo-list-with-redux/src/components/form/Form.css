.todo-form {
	margin-bottom: 50px;
}
