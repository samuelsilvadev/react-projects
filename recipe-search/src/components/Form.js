import React from 'react'
import PropTypes from 'prop-types';

import './Form.css';

const Form = ({ onSubmit }) => (
	<form className="search-form" onSubmit={onSubmit}>
		<input className="search-form__input" type="search" name="recipeName" />
		<button type="submit" className="search-form__button">Search</button>
	</form>
);

Form.propTypes = {
	onSubmit: PropTypes.func
};

export default Form;
