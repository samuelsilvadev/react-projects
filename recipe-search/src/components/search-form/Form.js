import React from 'react'
import PropTypes from 'prop-types';

import './Form.css';

const Form = ({ onSubmit }) => (
	<form className="search-form" onSubmit={onSubmit} data-test="search-form">
		<input className="search-form__input" type="search" name="recipeName" data-test="search-form__input" />
		<button type="submit" className="search-form__button" data-test="search-form__button">Search</button>
	</form>
);

Form.propTypes = {
	onSubmit: PropTypes.func
};

export default Form;
