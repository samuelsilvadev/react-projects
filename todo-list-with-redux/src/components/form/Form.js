import React from 'react';
import { connect } from 'react-redux';

import { addTodo } from './../../state/actions-creators';

import './Form.css';

export const Form = ({ addTodo }) => {

	const _handleSubmit = (event) => {
		event.preventDefault();
	
		const text = event.target.todo.value;
	
		addTodo(text);
	
		event.target.reset();
	};

	return (
		<form className='todo-form' onSubmit={_handleSubmit}>
			<input
				type='text'
				name='todo'
				required
				autoComplete='off'
				autoCapitalize='off'
			/>
			<button className='btn' type='submit'>
				Save
			</button>
		</form>
	);
};

const mapDispatchToProps = (dispatch) => ({
	addTodo: (text) => dispatch(addTodo(text)),
});

export default connect(null, mapDispatchToProps)(Form);
