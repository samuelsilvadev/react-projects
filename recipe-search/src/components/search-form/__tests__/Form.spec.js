import React from 'react';
import { shallow } from 'enzyme';

import Form from '../Form';

import { toTestId } from '@tests/utils';

describe('<Form />', () => {
	it('should render correctly', () => {
		const wrapper = shallow(<Form />);

		['search-form', 'search-form__input', 'search-form__button'].forEach(dataTest => {
			expect(wrapper.find(toTestId(dataTest)).exists()).toBe(true);
		});

		expect(wrapper).toMatchSnapshot();
	});

	it('should call the `onSubmit` when submit the form', () => {
		const onSubmit = jest.fn();
		const wrapper = shallow(<Form onSubmit={onSubmit} />);

		const form = wrapper.find(toTestId('search-form'));

		form.simulate('submit');

		expect(onSubmit).toHaveBeenCalledTimes(1);
	});
});
