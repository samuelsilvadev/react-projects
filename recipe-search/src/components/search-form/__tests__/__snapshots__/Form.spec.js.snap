// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render correctly 1`] = `
<form
  className="search-form"
  data-test="search-form"
>
  <input
    className="search-form__input"
    data-test="search-form__input"
    name="recipeName"
    type="search"
  />
  <button
    className="search-form__button"
    data-test="search-form__button"
    type="submit"
  >
    Search
  </button>
</form>
`;
